docs/tutorial/series/part_6.md

```diff
@@ -0,0 +1,140 @@
+│       │           ├── PersonInformation.class.php
+│       │           └── PersonInformationAction.class.php
+│       └── system
+│           ├── user
+│           │   └── activity
+│           │       └── event
+│           │           └── PersonInformationUserActivityEvent.class.php
+│           └── worker
+│               ├── PersonInformationRebuildDataWorker.class.php
+│               └── PersonRebuildDataWorker.class.php
+├── eventListener.xml
+├── language
+│   ├── de.xml
+│   └── en.xml
+└── objectType.xml
+```
+
+For all changes, please refer to the [source code on GitHub]({jinja{ config.repo_url }}tree/{jinja{ config.edit_uri.split("/")[1] }}/snippets/tutorial/tutorial-series/part-6).
+
+
+## User Activity Points
+
+The first step to support activity points is to register an object type for the `com.woltlab.wcf.user.activityPointEvent` object type definition for created person information and specify the default number of points awarded per piece of information:
+
+{jinja{ codebox(
+    language="xml",
+    title="objectType.xml",
+    contents="""
+<type>
+    <name>com.woltlab.wcf.people.information</name>
+    <definitionname>com.woltlab.wcf.user.activityPointEvent</definitionname>
+    <points>2</points>
+</type>
+    """
+) }}
+
+Additionally, the phrase `wcf.user.activityPoint.objectType.com.woltlab.wcf.people.information` (in general: `wcf.user.activityPoint.objectType.{objectType}`) has to be added.
+
+The activity points are awarded when new pieces are created via `PersonInformation::create()` using `UserActivityPointHandler::fireEvent()` and removed in `PersonInformation::create()` via `UserActivityPointHandler::removeEvents()` if pieces of information are deleted.
+
+Lastly, we have to add two components for updating data:
+First, we register a new rebuild data worker
+
+{jinja{ codebox(
+language="xml",
+title="objectType.xml",
+contents="""
+<type>
+    <name>com.woltlab.wcf.people.information</name>
+    <definitionname>com.woltlab.wcf.rebuildData</definitionname>
+    <classname>wcf\system\worker\PersonInformationRebuildDataWorker</classname>
+</type>
+"""
+) }}
+
+{jinja{ codebox(
+    title="files/lib/system/worker/PersonInformationRebuildDataWorker.class.php",
+    language="php",
+    filepath="tutorial/tutorial-series/part-6/files/lib/system/worker/PersonInformationRebuildDataWorker.class.php"
+) }}
+
+which updates the number of instances for which any user received person information activity points.
+(This data worker also requires the phrases `wcf.acp.rebuildData.com.woltlab.wcf.people.information` and `wcf.acp.rebuildData.com.woltlab.wcf.people.information.description`).
+
+Second, we add an event listener for `UserActivityPointItemsRebuildDataWorker` to update the total user activity points awarded for person information:
+
+{jinja{ codebox(
+    language="xml",
+    title="eventListener.xml",
+    contents="""
+<eventlistener name=\"execute@wcf\system\worker\\UserActivityPointItemsRebuildDataWorker\">
```

Comment:
Just wondering: Are the backslashes in front of the `"` actually required here? Does anything about the escape requirement change if we use a **r**aw string? The latter would also allow you to remove the duplicate backslash in front of the `u`.

```
r"""
foo
"""
```

Reply:
I was unable to get raw strings working.